File: cpp/861_翻转矩阵后的得分_medium.cpp
class Solution {
public:
    int matrixScore(vector<vector<int>>& A) {
        int m = A.size(), n = A[0].size();
        int ans = m*(1<<(n-1));
        for(int j=1; j<n; j++){
            int nCount = 0;
            for(int i=0; i<m; i++)
                if(A[i][0]==1)
                    nCount += A[i][j];
                else
                    nCount += (1-A[i][j]);
            int k = max(nCount, m-nCount);
            ans += k*(1<<(n-j-1));
        }
        return ans;
    }
};

File: python/861_翻转矩阵后的得分_medium.py
class Solution:
    def matrixScore(self, A: List[List[int]]) -> int:
        m, n = len(A), len(A[0])
        ans = m*(1<<(n-1))
        for j in range(1, n):
            nCount = 0
            for i in range(m):
                if A[i][0] == 1:
                    nCount += A[i][j]
                else:
                    nCount += (1-A[i][j])
            k = max(nCount, m-nCount)
            ans += k*(1<<(n-j-1))
        return ans